From 827f3b1465993703d6ea755d02a1578a510b79d5 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Wed, 17 Jun 2020 08:49:39 +0800
Subject: [PATCH 070/199] arm64: allwinner: dts: a64: enable HDMI port on
 Pinebook

Pinebook has a mini HDMI connector connected to the HDMI output of the
A64 SoC.

Enable it by activating the corresponding display pipeline and add
definition of the connector.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 .../dts/allwinner/sun50i-a64-pinebook.dts     | 22 +++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinebook.dts b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinebook.dts
index d427904417c2..72e4f2ec3f92 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinebook.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinebook.dts
@@ -48,6 +48,17 @@ lid_switch {
 		};
 	};
 
+	hdmi-connector {
+		compatible = "hdmi-connector";
+		type = "c";
+
+		port {
+			hdmi_con_in: endpoint {
+				remote-endpoint = <&hdmi_out_con>;
+			};
+		};
+	};
+
 	panel_edp: panel-edp {
 		compatible = "neweast,wjfh116008a";
 		backlight = <&backlight>;
@@ -135,6 +146,17 @@ &ehci1 {
 	status = "okay";
 };
 
+&hdmi {
+	hvcc-supply = <&reg_dldo1>;
+	status = "okay";
+};
+
+&hdmi_out {
+	hdmi_out_con: endpoint {
+		remote-endpoint = <&hdmi_con_in>;
+	};
+};
+
 &mixer0 {
 	status = "okay";
 };
-- 
2.28.0

